bin/reference_tools/update_bed.py

```diff
@@ -0,0 +1,450 @@
+        tmp_bed_path.unlink()
+
+
+def compare_clinvar(
+    new_clinvar: dict[str, ClinVarVariant],
+    old_clinvar: dict[str, ClinVarVariant],
+    final_bed_path: Path,
+    variant_padding: int,
+    out_dir: Path,
+    keep_tmp: bool,
+) -> tuple[list[list[str]], list[list[str]]]:
+    """
+    1. Check what variants are new / kept / removed among keys
+    2. Generate bed files for padded ClinVar variants to run "bedtools intersect"
+    3. Keep info about pathogenicity as the fourth column and variant key (chr:pos_ref_alt) in fifth
+    4. Return bed field lists
+    """
+
+    new_clinvar_keys = set(new_clinvar.keys())
+    old_clinvar_keys = set(old_clinvar.keys())
+
+    def write_tmp_bed(
+        path: Path, variant_dict: dict[str, ClinVarVariant], filter_keys: set[str]
+    ):
+        subset_dict = {
+            key: val
+            for (key, val) in variant_dict.items()
+            if key in filter_keys or len(filter_keys) == 0
+        }
+        bed_text = (
+            "\n".join(
+                [
+                    variant.get_padded_bed(variant_padding) + f"\t{var_key}"
+                    for (var_key, variant) in subset_dict.items()
+                ]
+            )
+            + "\n"
+        )
+        path.write_text(bed_text)
+
+    new_clinvar_tmp_bed = out_dir / "clinvar_new_python.bed"
+    write_tmp_bed(new_clinvar_tmp_bed, new_clinvar, set())
+    old_clinvar_tmp_bed = out_dir / "clinvar_old_python.bed"
+    clinvar_old_removed = old_clinvar_keys.difference(new_clinvar_keys)
+    write_tmp_bed(old_clinvar_tmp_bed, old_clinvar, clinvar_old_removed)
+
+    clinvar_in_common = new_clinvar_keys.intersection(old_clinvar_keys)
+    clinvar_new_added = new_clinvar_keys.difference(old_clinvar_keys)
+
+    new_to_add_bed_rows = get_bed_intersect(
+        str(new_clinvar_tmp_bed), str(final_bed_path)
+    )
+    old_to_remove_bed_rows = get_bed_intersect(
+        str(old_clinvar_tmp_bed), str(final_bed_path)
+    )
+    if not keep_tmp:
+        new_clinvar_tmp_bed.unlink()
+        old_clinvar_tmp_bed.unlink()
+
+    print(f"Clinvar in common between versions: {len(clinvar_in_common)}")
+    print(
+        f"Added new (unique targets): {len(clinvar_new_added)} ({len(new_to_add_bed_rows)})"
+    )
+    print(
+        f"Removed old (unique targets): {len(clinvar_old_removed)} ({len(old_to_remove_bed_rows)})"
+    )
+
+    return (new_to_add_bed_rows, old_to_remove_bed_rows)
+
+
+def get_bed_intersect(left_bed: str, right_bed: str) -> list[list[str]]:
+    not_in_bed_cmd = [
+        "bedtools",
+        "intersect",
+        "-a",
+        left_bed,
+        "-b",
+        right_bed,
+        "-v",
+    ]
+    result = subprocess.run(not_in_bed_cmd, capture_output=True, text=True, check=True)
+    intersected_regions = result.stdout.strip().splitlines()
+    # FIXME: This differs from the Perl script, and gives subtly differences in unique targets
+    # In this case 1270 vs 1274 (i.e. four are duplicated)
```

Comment:
Some more info please about what the subtle differences are and if it's the python or perl script that yields duplicates.

Reply:
Aha, I didn't clean this up. The difference is that the Perl yields some duplicates, and I run an extra `set` operation here removing them. I'll clarify

Reply:
Aha, or actually I skipped the setting here to cowardly align with the previous step. The ranges are merged later. I'll keep it as is for now for safety, and remove the comments